include/fastdds/rtps/history/IPayloadPool.h

```diff
@@ -22,27 +22,76 @@
 #include <fastdds/rtps/common/CacheChange.h>
 
 #include <cstdint>
-#include <functional>
+#include <memory>
 
 namespace eprosima {
 namespace fastrtps {
 namespace rtps {
 
+/**
+ * An interface for classes responsible of serialized payload management.
+ */
 class IPayloadPool
 {
 public:
 
     virtual ~IPayloadPool() = default;
 
+    /**
+     * @brief Get a serialized payload for a new sample.
+     * 
+     * This method will usually be called in one of the following situations:
+     * @li When a writer creates a new cache change
+     * @li When a reader receives the first fragment of a cache change
+     * 
+     * In both cases, the received @c size will be for the whole serialized payload.
+     * 
+     * @param [in]     size          Number of bytes required for the serialized payload
+     * @param [in,out] cache_change  Cache change to assign the payload to
+     * 
+     * @returns whether the operation succeeded or not
+     * 
+     * @post @li Field @c payload_owner of @c cache_change equals this
+     * @post @li Field @c serializedPayload.data points to a buffer of at least @c size bytes
+     * @post @li Field @c serializedPayload.max_size is greater than or equal to @c size
+     */
     virtual bool get_payload(
             uint32_t size,
-            const SampleIdentity& sample_identity,
             CacheChange_t& cache_change) = 0;
 
+    /**
+     * @brief Assign a serialized payload to a new sample.
+     * 
+     * This method will usually be called when a reader receives a whole cache change.
+     *
+     * @param [in]     data          Serialized payload received
+     * @param [in]     data_owner    Payload pool owning incoming data
+     * @param [in,out] cache_change  Cache change to assign the payload to
+     *
+     * @returns whether the operation succeeded or not
+     *
+     * @post @li Field @c payload_owner of @c cache_change equals this
+     * @post @li Field @c serializedPayload.data points to a buffer of at least @c data.length bytes
+     * @post @li Field @c serializedPayload.max_size is greater than or equal to @c data.length
+     */
```

Comment:
Can we restrict the postcondition even further and state that:
```
@li Field @c serializedPayload.data points to the same buffer pointed by @c data
```
and remove the `data.length` size postcondition, as it is redundant now? This change means we are specifically agreeing that we are not going to copy any data when this method is used.

Reply:
It would be up to the pool implementation to decide if copies should be done or not.